jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/yaml/YamlCloudFormationTemplate.kt

```diff
@@ -28,35 +30,41 @@ class YamlCloudFormationTemplate(template: YAMLFile) : CloudFormationTemplate {
 
     private val templateRoot = getTemplateRoot(template)
 
-    private fun getTemplateRoot(file: YAMLFile): YAMLMapping {
+    private fun getTemplateRoot(file: YAMLFile): YAMLMapping? {
         val documents = file.documents
         if (documents.size != 1) {
-            throw IllegalStateException(message("cloudformation.yaml.too_many_documents"))
+            LOG.debug { message("cloudformation.yaml.too_many_documents", file) }
+            return null
         }
 
-        return documents[0].topLevelValue as? YAMLMapping
-            ?: throw IllegalStateException(message("cloudformation.yaml.invalid_root_type"))
+        val topLevelValue = documents[0].topLevelValue
+        return if(topLevelValue is YAMLMapping) {
+            topLevelValue
+        } else {
+            LOG.debug { message("cloudformation.yaml.invalid_root_type") }
```

Comment:
should this message also include the `file` or some other identifying marker?

Reply:
Added file to the end